kyo-core/shared/src/main/scala/kyo/Channel.scala

```diff
@@ -158,6 +158,54 @@ object Channel:
           */
         def full(using Frame): Boolean < (Abort[Closed] & IO) = IO.Unsafe(Abort.get(self.full()))
 
+        private def emitChunks(maxChunkSize: Maybe[Int] = Maybe.Absent)(using
+            Tag[A],
+            Frame
+        ): Ack < (Emit[Chunk[A]] & Abort[Closed] & Async) =
+            if maxChunkSize.exists(_ <= 0) then Ack.Stop
+            else if maxChunkSize.exists(_ == 1) then
+                Loop(()): _ =>
+                    Channel.take(self).map: v =>
+                        Emit.andMap(Chunk(v)):
+                            case Ack.Stop => Loop.done(Ack.Stop)
+                            case _        => Loop.continue(())
+            else
+                val drainEffect = maxChunkSize.fold(Channel.drain(self))(m => Channel.drainUpTo(self)(m - 1))
+                Loop[Unit, Ack, Abort[Closed] & Async & Emit[Chunk[A]]](()): _ =>
+                    Channel.take(self).map: a =>
+                        drainEffect.map: chunk =>
+                            Emit.andMap(Chunk(a).concat(chunk)):
+                                case Ack.Stop => Loop.done(Ack.Stop)
+                                case _        => Loop.continue(())
+
+        /** Stream elements from channel, optionally specifying a maximum chunk size. In the absence of [[maxChunkSize]], chunk sizes will
+          * be limited only by channel capacity or the number of elements in the channel at a given time. (Chunks can still be larger than
+          * channel capacity.) Fails on channel closure.
+          *
+          * @param maxChunkSize
+          *   Maximum number of elements to take for each chunk
+          *
+          * @return
+          *   Asynchronous stream of elements in this channel
+          */
+        def stream(maxChunkSize: Maybe[Int] = Maybe.Absent)(using Tag[A], Frame): Stream[A, Abort[Closed] & Async] =
```

Comment:
I find it odd that the user needs to pass a `Maybe` to specify the max size. Could you use method overloads or an `Int.MaxValue` placeholder instead?